components/raftstore/src/store/peer.rs

```diff
@@ -2380,6 +2397,26 @@ impl Peer {
 
         poll_ctx.raft_metrics.propose.normal += 1;
 
+        // If applied index's term is differ from current raft's term, leader transfer
+        // must happened.
+        if self.get_store().applied_index_term() != self.term() {
+            return Err(box_err!(
+                "{} peer is not applied to current term, applied_term {}, current_term {}",
+                self.tag,
+                self.get_store().applied_index_term(),
+                self.term()
+            ));
+        }
+        if !self
+            .cmd_epoch_checker
+            .propose_check_epoch(&req, self.term())
```

Comment:
You can buffer these proposals in the `cmd_epoch_checker` and call callbacks when the admin command is finished to avoid backoff.